### PR TITLE
AuthMiddleware test

### PR DESCRIPTION
**Pull request by zhukovra**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #105 

I've made one change in `AuthMiddleware`. I think that without the change these lines is useless
https://github.com/yiisoft/yii-web/blob/a88d7fd6d6a2b9893e461134bf4472b70ddbe313/src/Auth/AuthMiddleware.php#L13-L15
https://github.com/yiisoft/yii-web/blob/a88d7fd6d6a2b9893e461134bf4472b70ddbe313/src/Auth/AuthMiddleware.php#L29
https://github.com/yiisoft/yii-web/blob/a88d7fd6d6a2b9893e461134bf4472b70ddbe313/src/Auth/AuthMiddleware.php#L42-L45

### CODE REVIEW

**Comment by demonking** [comment]
> https://github.com/yiisoft/yii-web/blob/a88d7fd6d6a2b9893e461134bf4472b70ddbe313/src/Auth/AuthMiddleware.php#L29
>
This seems right, because it doesn't return $this, it's a clone of $this : 

https://github.com/Nyholm/psr7/blob/master/src/ServerRequest.php#L143-L149

File: tests/Auth/AuthMiddlewareTest.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+use Yiisoft\Yii\Web\Auth\AuthMiddleware;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Yii\Web\User\IdentityInterface;
+
+class AuthMiddlewareTest extends TestCase
+{
+    /** @var ResponseFactoryInterface */
+    private $responseFactory;
+
+    /** @var AuthInterface */
```
Comment (demonking) [inline tests/Auth/AuthMiddlewareTest.php]: Should be like this:
```

 /**
  * @var AuthInterface|PHPUnit_Framework_MockObject_MockObject
  */

```

File: tests/Auth/AuthMiddlewareTest.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        $this->responseFactory = new Psr17Factory();
+        $this->authenticator = $this->createMock(AuthInterface::class);
+    }
+
+    /**
+     * @test
+     */
+    public function shouldAuthenticateAndSetAttribute()
+    {
+        $request = new ServerRequest('GET', '/');
```
Comment (demonking) [inline tests/Auth/AuthMiddlewareTest.php]: Don't know, but all other test have been written with mocks here :/ @samdark ?

File: tests/Auth/AuthMiddlewareTest.php (hunk 1 above)
Comment (zhukovra) [reply tests/Auth/AuthMiddlewareTest.php]: It is deprecated, so I change with actual class 

https://github.com/sebastianbergmann/phpunit/blob/fdc3245bc472bb1e23168b10aa0bd18716a58106/src/Framework/MockObject/MockObject.php#L21

**Comment by zhukovra** [comment]
> > https://github.com/yiisoft/yii-web/blob/a88d7fd6d6a2b9893e461134bf4472b70ddbe313/src/Auth/AuthMiddleware.php#L29
> 
> This seems right, because it doesn't return $this, it's a clone of $this :
> 
> https://github.com/Nyholm/psr7/blob/master/src/ServerRequest.php#L143-L149

It is part of PSR: https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/ServerRequestInterface.php#L235-L237

File: tests/Auth/AuthMiddlewareTest.php (hunk 2 above)
Comment (zhukovra) [reply tests/Auth/AuthMiddlewareTest.php]: Not all :) https://github.com/yiisoft/yii-web/pull/137#discussion_r341182954

Comment (samdark) [reply tests/Auth/AuthMiddlewareTest.php]: It's fine.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**